libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/Core/Binary/CDDL.hs

```diff
@@ -419,3 +463,224 @@ mkScriptInvalidHereafter =
         |This field specifies the right (excluded) endpoint b.
         |]
     $ "script_invalid_hereafter" =:~ grp [5, a slot]
+
+stake_credential :: Rule
+stake_credential = "stake_credential" =:= credential
+
+credential :: Rule
+credential = "credential" =:= arr [0, a addr_keyhash] / arr [1, a script_hash]
+
+drep_credential :: Rule
+drep_credential = "drep_credential" =:= credential
+
+committee_cold_credential :: Rule
+committee_cold_credential = "committee_cold_credential" =:= credential
+
+committee_hot_credential :: Rule
+committee_hot_credential = "committee_hot_credential" =:= credential
+
+port :: Rule
+port = "port" =:= VUInt `le` 65535
+
+ipv4 :: Rule
+ipv4 = "ipv4" =:= VBytes `sized` (4 :: Word64)
+
+ipv6 :: Rule
+ipv6 = "ipv6" =:= VBytes `sized` (16 :: Word64)
+
+dns_name64 :: Rule
+dns_name64 = "dns_name" =:= VText `sized` (0 :: Word64, 64 :: Word64)
+
+dns_name128 :: Rule
+dns_name128 = "dns_name" =:= VText `sized` (0 :: Word64, 128 :: Word64)
+
+url64 :: Rule
+url64 = "url" =:= VText `sized` (0 :: Word64, 64 :: Word64)
+
+url128 :: Rule
+url128 = "url" =:= VText `sized` (0 :: Word64, 128 :: Word64)
```

Comment:
This was introduced in cardano-ledger-conway, so let's move it there to avoid wrongful usage in prior eras